src/core/EventService.js

```diff
@@ -0,0 +1,46 @@
+export default class EventService {
+  constructor(socket) {
+    this._socket = socket;
+    this._refs = {};
+  }
+
+  _registerEvent(name, label, event, cb) {
+    const ref = this._socket.channel(name).on(event, cb);
+    this._refs[label + ':' + event] = ref;
+  }
+
+  _unregisterEvent(name, label, event) {
+    const ref = this._refs[label + ':' + event];
+    delete this._refs[label + ':' + event];
+    this._socket.channel(name).off(event, ref);
+  }
+
+  _cleanEventName(event) {
```

Comment:
rather than having two sets of event names, can we force everyone to just use one set?

Reply:
I agree, but this comes from the backend. There has been a major update to the backend as of last night which may include consistent event names.

Reply:
can we just use the single set of names the backend provides then? if you expect more changes to come from the backend, one way to do it would be to set up some constants, then only use the constants in this code, never hardcoded strings, so if the names change on the backend, you just have to change the value in this code once.

Reply:
That hasn't changed unfortunately so I think we are stuck until it's updated